libats/src/main/scala/com/advancedtelematic/libats/codecs/CirceCodecs.scala

```diff
@@ -63,6 +63,18 @@ trait CirceUri {
 
 object CirceUri extends CirceUri
 
+trait CirceNonEmptySet {
+
+  import cats.data.NonEmptySet.fromSetUnsafe
+
+  implicit def nonEmptySetDecoder[T](implicit dec: Decoder[Set[T]], ordering: Ordering[T]): Decoder[NonEmptySet[T]] =
+    dec.emapTry(xs => if (xs.isEmpty) Failure(DecodingFailure("No elements given for non-empty set.", Nil)) else Success(fromSetUnsafe[T](xs.to[SortedSet])))
```

Comment:
If we are moving it to libats we probably have to get rid of this `Nil` in `DecodingFailure`. To implement codec in the same way circe implements one for `NonEmptyList`